File: front/src/api/classroom.js
import request from '@/utils/request'
// 发送邮件
export const StudentLeave = (student, course, leaveType, reason, startTime, endTime) => {
  return request({
    url: '/classroom/student_leave/',
    method: 'POST',
    data: {
      'student': student,
      'course': course,
      'leave_type': leaveType,
      'reason': reason,
      'start_time': startTime,
      'end_time': endTime
    }
  })
}
// 获取全部课程
export const getAllCourse = () => {
  return request({
    url: '/classroom/student_leave/',
    method: 'GET'
  })
}

// 学生查看假条
export const StuCheckLeave = (searchTime) => {
  return request({
    url: '/classroom/student_check_leave/',
    method: 'GET',
    params: {
      'search_time': searchTime
    }
  })
}

// 老师获取自己所带课程的全部假条
export const TeaCheckLeave = (searchTime) => {
  return request({
    url: '/classroom/teacher_approval/',
    method: 'GET',
    params: {
      'search_time': searchTime
    }
  })
}

// 老师审批假条
export const TeacherApproval = (leaveId, status) => {
  return request({
    url: '/classroom/teacher_approval/',
    method: 'POST',
    data: {
      'leaveId': leaveId,
      'status': status
    }
  })
}

// 老师获取自己带的所有课程
export const CheckCourse = () => {
  return request({
    url: '/classroom/teacher_publish_notice/',
    method: 'GET'
  })
}

// 老师发布公告
export const publishNotice = (course, title, content) => {
  return request({
    url: '/classroom/teacher_publish_notice/',
    method: 'POST',
    data: {
      'course': course,
      'title': title,
      'content': content
    }
  })
}

// 老师查看公告
export const CheckNotice = (searchMsg) => {
  return request({
    url: '/classroom/teacher_check_notice/',
    method: 'GET',
    params: {
      'search_msg': searchMsg
    }
  })
}

// 返回所有学生
export const CheckStudents = (searchMsg) => {
  return request({
    url: '/classroom/teacher_pick_monitor/',
    method: 'GET',
    params: {
      'search_msg': searchMsg
    }
  })
}

// 返回选择班委
export const PickMonitor = (stuId) => {
  return request({
    url: '/classroom/teacher_pick_monitor/',
    method: 'POST',
    data: {
      'stu_id': stuId
    }
  })
}
// 学生查看班委
export const CheckMonitorStudent = () => {
  return request({
    url: '/classroom/student_check_monitor/',
    method: 'GET'
  })
}

// 管理员查看假条
export const CheckLeaveAdmin = () => {
  return request({
    url: '/classroom/admin_check_leave/',
    method: 'GET'
  })
}

// 管理员查看假条
export const CheckMonitorAdmin = () => {
  return request({
    url: '/classroom/admin_check_monitor/',
    method: 'GET'
  })
}

// 管理员查看公告
export const CheckNoticeAdmin = () => {
  return request({
    url: '/classroom/admin_check_notice/',
    method: 'GET'
  })
}

File: apps/core/urls/classroom.py
"""back URL Configuration

The `urlpatterns` list routes URLs to views. For more information please see:
    https://docs.djangoproject.com/en/4.1/topics/http/urls/
Examples:
Function views
    1. Add an import:  from my_app import views
    2. Add a URL to urlpatterns:  path('', views.home, name='home')
Class-based views
    1. Add an import:  from other_app.views import Home
    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
Including another URLconf
    1. Import the include() function: from django.urls import include, path
    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
"""
from django.urls import path

from apps.core.views.classroom.PickMonitor import PickMonitor, CheckMonitor
from apps.core.views.classroom.admin import CheckLeaveAdmin, CheckMonitorAdmin, CheckNoticeAdmin
from apps.core.views.classroom.approval import TeacherApproval
from apps.core.views.classroom.leave import StudentLeave, CheckLeaveStudent
from apps.core.views.classroom.publishNotice import PublishNotice, CheckNotice

urlpatterns = [
    path("student_leave/", StudentLeave.as_view(), name="学生请假"),
    path("student_check_leave/", CheckLeaveStudent.as_view(), name="学生查看假条"),
    path("student_check_monitor/", CheckMonitor.as_view(), name="学生查看班委"),
    path("teacher_approval/", TeacherApproval.as_view(), name="老师审批"),
    path("teacher_publish_notice/", PublishNotice.as_view(), name="发布公告"),
    path("teacher_check_notice/", CheckNotice.as_view(), name="老师查看公告"),
    path("teacher_pick_monitor/", PickMonitor.as_view(), name="老师设立班委"),
    path("admin_check_leave/", CheckLeaveAdmin.as_view(), name="管理员查看假条"),
    path("admin_check_monitor/", CheckMonitorAdmin.as_view(), name="管理员查看假条"),
    path("admin_check_notice/", CheckNoticeAdmin.as_view(), name="管理员查看公告"),
]
